optimum/neuron/utils/hub_neuronx_cache.py

```diff
@@ -364,25 +417,151 @@ def get_hub_cached_entries(
+
+    return entry_config, target_entry_config, neuron_config
+
+
+def get_multimodels_configs_from_hub(model_id):
+    api = HfApi()
+    repo_files = api.list_repo_files(model_id)
+    config_pattern = "/config.json"
+    config_files = [path for path in repo_files if config_pattern in path]
+    lookup_configs = {}
+    with TemporaryDirectory() as tmpdir:
+        for model_path in config_files:
+            local_path = api.hf_hub_download(model_id, model_path, local_dir=tmpdir)
+            with open(local_path) as f:
+                entry_config = json.load(f)
+                white_list = CACHE_WHITE_LIST
                 for param in white_list:
                     entry_config.pop(param, None)
-                    target_entry.config.pop(param, None)
-                if entry_config == target_entry.config:
-                    model_entries.append(neuron_config)
-    return model_entries
+                lookup_configs[model_path.split("/")[-2]] = entry_config
+
+    if "unet" in lookup_configs:
+        lookup_configs["model_type"] = "stable-diffusion"
+    return lookup_configs
+
+
+def exclude_white_list_from_config(
+    config: Dict, white_list: Optional[List] = None, neuron_white_list: Optional[List] = None
+):
+    if white_list is None:
+        white_list = CACHE_WHITE_LIST
+
+    if neuron_white_list is None:
+        neuron_white_list = NEURON_CONFIG_WHITE_LIST
+
+    for param in white_list:
+        config.pop(param, None)
+
+    for param in neuron_white_list:
+        config["neuron"].pop(param, None)
+
+    return config
+
+
+# Only applied on traced TorchScript models
+def build_cache_config(
+    configs: Union[PretrainedConfig, Dict[str, PretrainedConfig]],
+    white_list: Optional[List] = None,
+    neuron_white_list: Optional[List] = None,
+):
+    clean_configs = {}
+    no_check_components = [
+        "vae",
+        "vae_encoder",
+        "vae_decoder",
+    ]  # Exclude vae configs from stable diffusion pipeline since it's complex and not mandatory
+    if isinstance(configs, PretrainedConfig):
+        configs = {"model": configs}
+    for name, config in configs.items():
+        if name in no_check_components:
+            continue
+        config = copy.deepcopy(config).to_diff_dict() if isinstance(config, PretrainedConfig) else config
+        config = exclude_white_list_from_config(config, white_list, neuron_white_list)
+        clean_configs[name] = config
+
+    if len(clean_configs) > 1:
+        if "unet" in configs:
+            # stable diffusion
+            clean_configs["model_type"] = "stable-diffusion"
+        else:
+            # seq-to-seq
+            clean_configs["model_type"] = next(iter(clean_configs.values()))["model_type"]
+
+        return clean_configs
+    else:
+        return next(iter(clean_configs.values()))
+
+
+def cache_aot_neuron_artifacts(neuron_dir: Path, cache_config_hash: str):
+    cache_repo_id = load_custom_cache_repo_name_from_hf_home()
```

Comment:
1. This is where you should call the context.
2. No need to reevaluate cache_repo_id, as you use the default value.
3. Inside the context, you should call directly `create_compile_cache` (which has been patched to _create...)

Reply:
Applied the first two suggestions, and I would keep with `create_hub_compile_cache_proxy()` instead of `hf_create_compile_cache` since it asks for the required positional argument: 'cache_url'.